### PR TITLE
fix missing attribs with dirct module execution

### LINKED ISSUE #53872: Directly runnig module doesn't work with fetch_url() function within module
##### SUMMARY

I'm trying to run my module locally using next instructions https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html#exercising-module-code-locally

And I get an error.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

Any module containing fetch_url() function from ansible.module_utils.urls
In my case it is my module I'm developing https://github.com/okgolove/ansible/blob/sentry_fixes/lib/ansible/modules/web_infrastructure/sentry_project.py

##### ANSIBLE VERSION

```paste below
ansible 2.8.0.dev0
  config file = None
  configured module search path = [u'/Users/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/user/git/ansible/lib/ansible
  executable location = /Users/user/git/ansible/bin/ansible
  python version = 2.7.15 (default, Oct  2 2018, 11:47:18) [GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.2)]
```

##### CONFIGURATION

args.json
```paste below
{
    "ANSIBLE_MODULE_ARGS": {
        "api_token": "abcxdz",
        "organization": "develop",
        "project_name": "backendtest",
        "project_slug": "backendtest",
        "state": "present",
        "team": "develop",
        "url": "https://sentry.io"
    }
}
```

##### OS / ENVIRONMENT

MacOS 10.14.3, Python 2.7.15

##### STEPS TO REPRODUCE

```pipenv shell && source hacking/env-setup && python lib/ansible/modules/web_infrastructure/sentry_project.py args.json```

##### EXPECTED RESULTS

Module runs properly.

##### ACTUAL RESULTS

And I get an error:

```Traceback (most recent call last):
  File "lib/ansible/modules/web_infrastructure/sentry_project.py", line 159, in <module>
    main()
  File "lib/ansible/modules/web_infrastructure/sentry_project.py", line 118, in main
    if is_project_exists(project_name, organization, team, api_token):
  File "lib/ansible/modules/web_infrastructure/sentry_project.py", line 110, in is_project_exists
    method="GET")
  File "/Users/user/git/ansible/lib/ansible/module_utils/urls.py", line 1327, in fetch_url
    tempfile.tempdir = module.tmpdir
  File "/Users/user/git/ansible/lib/ansible/module_utils/basic.py", line 851, in tmpdir
    basedir = os.path.expanduser(os.path.expandvars(self._remote_tmp))
AttributeError: 'AnsibleModule' object has no attribute '_remote_tmp'
```

### PR DESCRIPTION
**Pull request by bcoca**
now always populate attributes, even if data is not passed in.
fixes #53872
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

module_utils/

### CODE REVIEW

**Comment by okgolove** [comment]
@bcoca what a strange thing. I remember that it worked and now it has stopped. 
I've applied a patch from this PR, running module directly works, but running unit tests doesn't work:

```
(ansible-sYHAY5jb) » user@okgolove-mac  ~/git/ansible(sentry_tests) $ ansible-test units --python 2.7 sentry
Unit test with Python 2.7
==================================================================== test session starts =====================================================================
platform darwin -- Python 2.7.10, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /Users/user/git/ansible, inifile: tox.ini
plugins: xdist-1.27.0, forked-1.0.2
gw0 [1] / gw1 [1] / gw2 [1] / gw3 [1]
F
========================================================================== FAILURES ==========================================================================
_____________________________________________________________________ test_main_function _____________________________________________________________________
[gw1] darwin -- Python 2.7.10 /tmp/python-8ad_64-ansible/python
def test_main_function():
        set_module_args({
            'api_token': '1234567890abcdwxyz',
            'organization': 'example',
            'project_name': 'backend',
            'project_slug': 'backend',
            'state': 'present',
            'team': 'senior',
            'url': 'localhost'
        })
>       sentry_project.main()

test/units/modules/web_infrastructure/test_sentry.py:23:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
lib/ansible/modules/web_infrastructure/sentry_project.py:144: in main
    if does_project_exist(project_name, organization, team, api_token):
lib/ansible/modules/web_infrastructure/sentry_project.py:136: in does_project_exist
    method="GET")
lib/ansible/module_utils/urls.py:1327: in fetch_url
    tempfile.tempdir = module.tmpdir
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ansible.module_utils.basic.AnsibleModule object at 0x10f24b4d0>

    @property
    def tmpdir(self):
        # if _ansible_tmpdir was not set and we have a remote_tmp,
        # the module needs to create it and clean it up once finished.
        # otherwise we create our own module tmp dir from the system defaults
        if self._tmpdir is None:
            basedir = None

>           if self._remote_tmp is not None:
E           AttributeError: 'AnsibleModule' object has no attribute '_remote_tmp'

lib/ansible/module_utils/basic.py:850: AttributeError
----------------------------------- generated xml file: /Users/user/git/ansible/test/results/junit/python2.7-units.xml -----------------------------------
================================================================== short test summary info ===================================================================
FAILED test/units/modules/web_infrastructure/test_sentry.py::test_main_function
================================================================== 1 failed in 1.98 seconds ==================================================================
ERROR: Command "pytest --boxed -r a -n auto --color yes --junit-xml test/results/junit/python2.7-units.xml test/units/modules/web_infrastructure/test_sentry.py" returned exit status 1.
```


test_sentry.py:
```
import json
import pytest
from ansible.module_utils import basic
from ansible.module_utils._text import to_bytes
import ansible.modules.web_infrastructure.sentry_project as sentry_project


def set_module_args(args):
    args = json.dumps({'ANSIBLE_MODULE_ARGS': args})
    basic._ANSIBLE_ARGS = to_bytes(args)


def test_main_function():
    set_module_args({
        'api_token': '1234567890abcdwxyz',
        'organization': 'example',
        'project_name': 'backend',
        'project_slug': 'backend',
        'state': 'present',
        'team': 'senior',
        'url': 'localhost'
    })
    sentry_project.main()

```

File: test/integration/targets/run_modules/runme.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+set -eux
+
+# test running module directly
+library/test.py args.json
```
Comment (mattclay) [inline test/integration/targets/run_modules/runme.sh]:
```suggestion
python.py library/test.py args.json
```

Comment (bcoca) [reply test/integration/targets/run_modules/runme.sh]: @mattclay trying to reproduce the original issue, which was 'direct module execution', I can add that as additional test, but would like the existing to stay

Comment (mattclay) [reply test/integration/targets/run_modules/runme.sh]: The proposed change is still direct module execution. It does not go through Ansible. It does make sure the correct python interpreter is used though.

File: test/integration/targets/run_modules/aliases [hunk 2]
```diff
@@ -0,0 +1,3 @@
+shippable/posix/group1
+shippable/posix/group3
+shippable/posix/group4
```
Comment (mattclay) [inline test/integration/targets/run_modules/aliases]: Only one group should be selected.

**Pull request merged**